src/components/type-group/type-option/TypeOption.js

```diff
@@ -0,0 +1,55 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import styles from './TypeOption.css';
+
+class TypeOption extends Component {
+    render() {
+        const { label, groupName, children, selected, badge: Badge, selectable } = this.props;
+        const labelClasses = classNames(styles.label, selectable && styles.selectable);
+
+        return (
+            <div className={ styles.container }>
+                <label className={ labelClasses }>
+                    <input className={ styles.input }
+                        type="radio"
+                        name={ groupName }
+                        checked={ selected }
+                        disabled={ !selectable }
+                        onChange={ this.handleInputChange } />
+                    <div className={ styles.circle }>
+                        { Badge &&
+                            <span className={ styles.badge }>
+                                <Badge />
+                            </span> }
+                        { children }
+                    </div>
+                    <span>{ label }</span>
+                </label>
+            </div>
+        );
+    }
+
+    handleInputChange = () => {
+        this.props.onSelect(this.props.id);
+    };
+}
+
+TypeOption.propTypes = {
+    groupName: PropTypes.string,
+    selectable: PropTypes.bool,
+    selected: PropTypes.bool,
+    children: PropTypes.node,
+    onSelect: PropTypes.func,
+    label: PropTypes.string,
+    badge: PropTypes.func,
```

Comment:
Shouldn’t the badge be a element instead of a component? This allows to inject props like so `badge={ <SomeBadge foo=“bar” /> }`?

Reply:
I got your point 👍